inverse_covariance/profiling/graphs.py

```diff
@@ -0,0 +1,188 @@
+import numpy as np 
+import scipy as sp
+
+
+def lattice(prng, n_features, alpha, random_sign=False, low=0.3, high=0.7):
+    """Returns the adjacency matrix for a lattice network. 
+    
+    The resulting network is a Toeplitz matrix with random values summing 
+    between -1 and 1 and zeros along the diagonal. 
+    
+    The range of the values can be controlled via the parameters low and high.  
+    If random_sign is false, all entries will be negative, otherwise their sign
+    will be modulated at random with probability 1/2.
+
+    Each row has maximum edges of np.ceil(alpha * n_features).
+
+    Parameters
+    -----------        
+    n_features : int 
+
+    alpha : float (0, 1) 
+        The complexity / sparsity factor.
+
+    random sign : bool (default=False)
+        Randomly modulate each entry by 1 or -1 with probability of 1/2.
+
+    low : float (0, 1) (default=0.3)
+        Lower bound for np.random.RandomState.uniform before normalization.
+
+    high : float (0, 1) > low (default=0.7)
+        Upper bound for np.random.RandomState.uniform before normalization.
+    """
+    adj = np.zeros((n_features, n_features))
+    row = np.zeros((n_features,))
+    degree = int(1 + np.round(alpha * n_features / 2.))
+
+    if random_sign:
+        sign_row = (-1.0 * np.ones(degree) + 
+                    2 * (prng.uniform(low=0, high=1, size=degree) > .5))
+    else:
+        # QUESTION to M:
+        # Why is the default all negatives?
+        sign_row = -1.0 * np.ones(degree)
+    
+    # populate shifted by 1 to avoid diagonal
+    row[1: 1 + degree] = sign_row * prng.uniform(low=low, high=high, size=degree)
+
+    # QUESTION to M:
+    # Why is this normalized this way (can sum between 1 and -1)?
+    # What if the row sums to 0 (unlikely, but could be)? 
+    row /= np.abs(np.sum(row))  
+
```

Comment:
@jasonlaska row values needs to be either only negative valued or both negative/positive values as we are creating true precision rather than covariance matrix. (all negative precision values corresponding to positively correlated entries in covariance).  If sum ends up being negative, I still want row to be negative valued not positive valued. 

To address the issues you bring up maybe use `row /=np.abs(np.sum(row)` only `np.sum(row)` is non-zero otherwise maybe `row /=np.abs(np.sum(np.abs(row)`? I don't like the latter scenario, but maybe it will do for now.